extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterConfig.java

```diff
@@ -47,19 +105,56 @@
       @JsonProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG) String bootstrapServers,
-      @JsonProperty("metric.topic") String metricTopic,
-      @JsonProperty("alert.topic") String alertTopic,
+      @Nullable @JsonProperty("event.types") Set<EventType> eventTypes,
+      @Nullable @JsonProperty("metric.topic") String metricTopic,
+      @Nullable @JsonProperty("alert.topic") String alertTopic,
       @Nullable @JsonProperty("request.topic") String requestTopic,
+      @Nullable @JsonProperty("segmentMetadata.topic") String segmentMetadataTopic,
+      @Nullable @JsonProperty("segmentMetadata.topic.format") SegmentMetadataTopicFormat segmentMetadataTopicFormat,
       @JsonProperty("clusterName") String clusterName,
       @JsonProperty("producer.config") @Nullable Map<String, String> kafkaProducerConfig
   )
   {
     this.bootstrapServers = Preconditions.checkNotNull(bootstrapServers, "bootstrap.servers can not be null");
-    this.metricTopic = Preconditions.checkNotNull(metricTopic, "metric.topic can not be null");
-    this.alertTopic = Preconditions.checkNotNull(alertTopic, "alert.topic can not be null");
-    this.requestTopic = requestTopic;
+    validateAndSetEventTypes(eventTypes, requestTopic);
+    validateAndSetSegmentMetadataTopicFormat(segmentMetadataTopicFormat);
+
+    this.metricTopic = this.eventTypes.contains(EventType.METRICS) ? Preconditions.checkNotNull(metricTopic, "metric.topic can not be null") : null;
+    this.alertTopic = this.eventTypes.contains(EventType.ALERTS) ? Preconditions.checkNotNull(alertTopic, "alert.topic can not be null") : null;
+    this.requestTopic = this.eventTypes.contains(EventType.REQUESTS) ? Preconditions.checkNotNull(requestTopic, "request.topic can not be null") : null;
+    this.segmentMetadataTopic = this.eventTypes.contains(EventType.SEGMENT_METADATA) ? Preconditions.checkNotNull(segmentMetadataTopic, "segmentMetadata.topic can not be null") : null;
     this.clusterName = clusterName;
     this.kafkaProducerConfig = kafkaProducerConfig == null ? ImmutableMap.of() : kafkaProducerConfig;
+
+    // Metadata in non-JSON format must go their own topic
+    if (this.segmentMetadataTopic != null && !SegmentMetadataTopicFormat.JSON.equals(this.segmentMetadataTopicFormat)) {
+      this.segmentMetadataTopic = this.segmentMetadataTopic + "-" + this.segmentMetadataTopicFormat;
+    }
+  }
+
+  private void validateAndSetEventTypes(Set<EventType> eventTypes, String requestTopic)
+  {
+    // Unless explicitly overridden, kafka emitter will always emit metrics and alerts
+    if (eventTypes == null) {
+      this.eventTypes = DEFAULT_EVENT_TYPES;
+      // To maintain backwards compatibility, if eventTypes is not set, then requests are sent out or not
+      // based on the `request.topic` config
+      if (requestTopic != null) {
+        this.eventTypes.add(EventType.REQUESTS);
+      }
+    } else {
+      this.eventTypes = new HashSet<>();
+      this.eventTypes.addAll(eventTypes);
```

Comment:
any reason we are making a copy of the set here? If we are trying to avoid the original set being mutated for some reason, you could use ImmutableSet.copyOf(eventTypes), but in this case it might be overkill, since Jackson will not mutate the set it returned.